src/lib.rs

```diff
@@ -284,6 +280,11 @@ impl UUID {
             uuid: Uuid::from_u128(int),
         })
     }
+
+    fn __reduce__<'a>(&self, py: Python<'a>) -> PyResult<&'a PyTuple> {
```

Comment:
I think it’s simpler to use ‘__getnewargs__’ as is explained here:
https://github.com/PyO3/pyo3/issues/100#issuecomment-1588641927

Reply:
Thx, improved that one